libs/ledger/src/state_adapter.cpp

```diff
@@ -143,11 +144,11 @@ StateAdapter::Status StateAdapter::Exists(std::string const &key)
  * @param key The input key to be converted
  * @return The generated resource address for this key
  */
-ResourceAddress StateAdapter::CreateAddress(Identifier const &scope, ConstByteArray const &key)
+ResourceAddress StateAdapter::CreateAddress(ConstByteArray const &scope, ConstByteArray const &key)
 {
-  FETCH_LOG_DEBUG("StateAdapter", "Creating address for key: ", key, " scope: ", scope.full_name());
+  FETCH_LOG_DEBUG("StateAdapter", "Creating address for key: ", key, " scope: ", scope);
 
-  return ResourceAddress{scope.full_name() + ".state." + key};
+  return ResourceAddress{scope + ".state." + key};
```

Comment:
This is a good place to encode, I think. I'm not sure if scope can be equal to fetch.token here, so you may need to check for that

Reply:
scope can be token as well. So I don't think this is a good place for encoding.